src/components/ItemList.js

```diff
@@ -1,27 +1,38 @@
 import { serverTimestamp, Timestamp, updateDoc, doc } from 'firebase/firestore';
-import { db } from '../lib/firebase.js';
 import { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { calculateEstimate } from '@the-collab-lab/shopping-list-utils';
+import { db } from '../lib/firebase.js';
 import useFirebaseSnapshot from '../hooks/useFirebaseSnapshot.js';
 import cleanData from '../utils/cleanData.js';
-import { Link } from 'react-router-dom';
 
 const ItemList = () => {
   const { docs, loading } = useFirebaseSnapshot();
   const [searchInput, setSearchInput] = useState('');
   const [filteredResults, setFilteredResults] = useState('');
 
-  const handleChecked = async (id) => {
+  const handleChecked = async (id, item) => {
+    const daysSinceLastPurchased = item.data['last purchased']
+      ? Math.round((Timestamp.now() - item.data['last purchased']) / 86400)
+      : 0;
+
     //console.log used for testing checked item and manipulating last purchased time in db
-    console.log(id);
+    // console.log(id);
     const docRef = doc(db, 'shopping-list', id);
     await updateDoc(docRef, {
+      'estimated purchase interval': calculateEstimate(
+        item.data['estimated purchase interval'],
+        daysSinceLastPurchased,
+        item.data['total purchases'],
+      ),
       'last purchased': serverTimestamp(),
+      'total purchases': (item.data['total purchases'] += 1),
     });
   };
 
   const within24Hours = (item) => {
     // Seconds in 24 hours
-    // 60 x 60 x 24 = 86400
+    // 60 x 60 x 24 = 86400å
```

Comment:
is the character at the end of the line an unintended artifact?

Reply:
yup, removed.